packages/mdc-theme/_variables.scss

```diff
@@ -140,3 +140,22 @@ $mdc-theme-property-values: (
   text-disabled-on-dark: map-get(map-get($mdc-theme-text-colors, light), disabled),
   text-icon-on-dark: map-get(map-get($mdc-theme-text-colors, light), icon)
 );
+
+// Looks up the given theme property from `mdc-theme-prop` and returns its value if it exists.
+// Otherwise, returns the property verbatim.
+//
+// This is mainly useful in situations where `mdc-theme-prop` cannot be used directly (e.g., `box-shadow`).
+//
+// Examples:
+//
+// 1. mdc-theme-prop-value(primary) => "#3f51b5"
+// 2. mdc-theme-prop-value(blue) => "blue"
+//
+// NOTE: This function must be defined in _variables.scss instead of _functions.scss to avoid circular imports.
+@function mdc-theme-prop-value($property) {
```

Comment:
Should we make this mirror the functionality of mdc-theme-prop where it checks if it's a color and treats that as a passthrough, otherwise it checks the map and will actively cause an error if the key doesn't exist? I'm a bit concerned this could lead to silent failures as written.

Also, I guess there's no reasonable way for us to factor CSS custom properties into this function; maybe we should also note that as a difference between this and the mixin?

Reply:
Done. Good call!